pkg/workflowengine/impl/prepare_execution.go

```diff
@@ -40,6 +40,21 @@ func addPermissions(auth *admin.AuthRole, roleNameKey string, flyteWf *v1alpha1.
 	}
 }
 
+func addPermissionsFromSecurtiyCtx(securityCtx *core.SecurityContext, roleNameKey string, flyteWf *v1alpha1.FlyteWorkflow) {
```

Comment:
what if you just pass the security context to addPermissions and have it prefer values from non-empty securitycontexts so that permissions handling is done in one place?

Reply:
Done .Added a wrapper addPermissions which accepts both authRole and securityCtx